ksp_plugin/plugin.cpp

```diff
@@ -380,15 +380,19 @@ void Plugin::InsertOrKeepVessel(GUID const& vessel_guid,
   CHECK(!initializing_);
   not_null<Celestial const*> parent =
       FindOrDie(celestials_, parent_index).get();
-  GUIDToOwnedVessel::iterator it;
-  std::tie(it, inserted) =
+  {
+    bool& should_insert = inserted;
+    should_insert = !Contains(vessels_, vessel_guid);
+    if (should_insert) {
       vessels_.emplace(vessel_guid,
                        make_not_null_unique<Vessel>(vessel_guid,
                                                     vessel_name,
                                                     parent,
                                                     ephemeris_.get(),
                                                     prediction_parameters_));
-  not_null<Vessel*> const vessel = it->second.get();
+    }
+  }
+  not_null<Vessel*> const vessel = FindOrDie(vessels_, vessel_guid).get();
```

Comment:
Isn't it a bit ugly that we are doing a lookup here when we just did a lookup or an insert.  I'd use vanilla STL:
```
auto it = vessels_.find(...);
if (it == vessels_.end()) {
   std::tie(...) = vessels_.emplace(...);
}
vessel = it->second;
```